### PR TITLE
MWPW-161191:  Checkout cli parameter does not get rendered correctly

### PR DESCRIPTION
**Pull request by Axelcureno**
Fixes CLI parameter in checkout links in milo consumer projects like DC, CC.

Resolves: [MWPW-161191](https://jira.corp.adobe.com/browse/MWPW-161191)

Actual test URL (needs to be tested from milo consumer project): https://main--dc--adobecom.hlx.live/acrobat/features?milolibs=MWPW-161191--milo--adobecom

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/docs/library/kitchen-sink/merch-card?martech=off
- After: https://MWPW-161191--milo--adobecom.hlx.page/docs/library/kitchen-sink/merch-card?martech=off

### CODE REVIEW

**Review by Roycethan — CHANGES_REQUESTED** [review]
@Axelcureno Its breaking Pricing and Checkout CTA's (label and commerce url) at CC pages:
https://main--cc--adobecom.hlx.live/creativecloud?milolibs=MWPW-161191--milo--adobecom
image

cc @yesil  @npeltier @afmicka @3ch023

File: libs/blocks/merch/merch.js [hunk 1]
```diff
@@ -629,7 +646,7 @@ export async function getCheckoutContext(el, params) {
 
   return {
     ...context,
-    checkoutClientId,
+    'checkout-client-id': checkoutClientId,
```
Comment (npeltier) [inline libs/blocks/merch/merch.js:649]: i'm not sure why we are doing this: mapping in & out from kebab case is done for having well formed attributes but before and after camel case is used iirc

Comment (Axelcureno) [reply libs/blocks/merch/merch.js]: true, I have removed it.

**Comment by Roycethan** [comment]
> @Axelcureno Its breaking Pricing and Checkout CTA's (label and commerce url) at CC pages: https://main--cc--adobecom.hlx.live/creativecloud?milolibs=MWPW-161191--milo--adobecom image
> 
> cc @yesil @npeltier @afmicka @3ch023

@Axelcureno  Fixed now with latest commit. Thanks

**Review by Roycethan — APPROVED** [review]

**Comment by Roycethan** [comment]
@Axelcureno  Verified, Need 2 approvals for Ready for stage label, and if there any changes a smoke test. Thanks

File: libs/blocks/merch/merch.js [hunk 2]
```diff
@@ -560,6 +560,23 @@ export async function initService(force = false, attributes = {}) {
...
+    'checkout-client-id',
+    'allow-override',
+  ];
+
+  extraAttrs.forEach((attr) => {
+    const camelCaseAttr = attr.replace(/-([a-z])/g, (g) => g[1].toUpperCase());
+    // eslint-disable-next-line no-prototype-builtins
+    if (commerce?.hasOwnProperty(camelCaseAttr)) {
+      const value = commerce[camelCaseAttr];
+      delete commerce[camelCaseAttr];
```
Comment (yesil) [inline libs/blocks/merch/merch.js:576]: I think we should not alter `commerce` object.

File: libs/blocks/merch/merch.js [hunk 3]
```diff
@@ -560,6 +560,23 @@ export async function initService(force = false, attributes = {}) {
     fetchCheckoutLinkConfigs.promise = undefined;
   }
   const { commerce, env: miloEnv, locale: miloLocale } = getConfig();
+
+  const extraAttrs = [
+    'checkout-workflow-step',
+    'force-tax-exclusive',
+    'checkout-client-id',
+    'allow-override',
```
Comment (yesil) [inline libs/blocks/merch/merch.js:568]: `allow-override` is not controlled by the consumer but by merch.js, and it is set only for non prod domains.
we should remove it from the list.

File: libs/blocks/merch/merch.js (hunk 2 above)
Comment (npeltier) [reply libs/blocks/merch/merch.js:576]: those attributes could be added through attributes object

**Review by npeltier — APPROVED** [review]
i generally agree with @yesil feedbacks, but this can be treated afterward

**Review by yesil — APPROVED** [review]
my feedback can be addressed later.

**Pull request merged**